IBM/carbon-addons-iot-react
# 1065 action buttons visual defects when parent is expanded

Closes #1065

**Summary**

> The new fix we did for actions buttons has a small defect when a row is expanded. Seems like action buttons are always show when expanded but the background color is only on hover. So you can see the rows behind it.

Applied fix so that actions are not automatically visible once the roe is expanded. Actions are only visible if the user hovers the row or if the action overflow menu is showing.

It was not possible to add any meaningful unit tests for this fix since it relies on hover and css rather than props and js.

**Change List (commits, features, bugs, etc)**
- Row actions not showing unless row is hovered, even if row is expanded
- Made sure the row actions are visible even when not hovering if the action overflowmenu is open
- Fixed color of row-actions-container__background depending on color of the row (normal, hover, selected)
- Added unit test to compensate for fallen threshold as a consequence of the small refactoring

**Acceptance Test (how to verify the PR)**
 1. to the PRs stories and choose [watson-iot-table--stateful-example-with-expansion-and-column-resize](https://deploy-preview-1102--carbon-addons-iot-react.netlify.app/?path=/story/watson-iot-table--stateful-example-with-expansion-and-column-resize)
2. Hover the rows to make sure the row actions are showing
3. Expand the row and do not hover the original row, make sure the row actions are not showing.
4. Hover the the original row to verify that the row actions are present

_Additional testing for refactored/improved code_
5. select the row using the checkbox, make sure the row actions background is of the same color as the selected row
6. Combine the above tests with clicking the row-action button with overflow menu and hover the menu alternatives. Make sure the row actions stay visible and with the correct background color.
